material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Application Library Engineering Group.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Ripple } from './ripple/ripple'
+import { FabExtendedModel } from './fabUtils/fab_extended_model'
+
+@Component
+export struct FAB_extended {
+  @State model: FabExtendedModel.Model = undefined
+  @State show_ripple: boolean = false
+  onClick: (event?: ClickEvent) => void
+
+  build() {
+    Row() {
+      if (this.model.getFabIcon()) {
```

Comment:
Add validation on model to check if its not null and not undefined... add similar validation for all required variables in all files